programs/server/Server.cpp

```diff
@@ -1822,9 +1822,21 @@ try
             else
                 LOG_INFO(log, "Closed all listening sockets.");
 
-            /// Killing remaining queries.
+            /// Killing all remaining queries.
             if (!server_settings.shutdown_wait_unfinished_queries)
+            {
                 global_context->getProcessList().killAllQueries();
+            }
+            /// Killing only selected queries.
+            else
+            {
+                if (!server_settings.shutdown_wait_unfinished_query_kind.value.empty())
+                    for (auto & query : global_context->getProcessList().processes)
+                    {
+                        if (query->getInfo().query_kind != server_settings.shutdown_wait_unfinished_query_type)
```

Comment:
I don't like that you can set only one query_kind here. Maybe provide some levels?
`1 - insert`, `2 - insert,select`,...

Reply:
We support multiple values for settings like `join_algorithm='direct, default'`
Why not have here the same?

Reply:
Thank you for the reference. I am checking it out.